src/main/java/tech/blacksource/blacknectar/service/algorithms/GooglePlacesStoreSearchAlgorithm.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2017 BlackSource, LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tech.blacksource.blacknectar.service.algorithms;
+
+import java.util.List;
+import javax.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sir.wellington.alchemy.collections.lists.Lists;
+import tech.aroma.client.Aroma;
+import tech.aroma.client.Urgency;
+import tech.blacksource.blacknectar.service.exceptions.OperationFailedException;
+import tech.blacksource.blacknectar.service.stores.Store;
+import tech.redroma.google.places.GooglePlacesAPI;
+import tech.redroma.google.places.data.Location;
+import tech.redroma.google.places.data.Place;
+import tech.redroma.google.places.exceptions.GooglePlacesException;
+import tech.redroma.google.places.requests.NearbySearchRequest;
+import tech.sirwellington.alchemy.annotations.designs.patterns.StrategyPattern;
+
+import static tech.blacksource.blacknectar.service.stores.Store.validStore;
+import static tech.sirwellington.alchemy.annotations.designs.patterns.StrategyPattern.Role.CONCRETE_BEHAVIOR;
+import static tech.sirwellington.alchemy.arguments.Arguments.checkThat;
+import static tech.sirwellington.alchemy.arguments.assertions.Assertions.notNull;
+
+/**
+ *
+ * @author SirWellington
+ */
+@StrategyPattern(role = CONCRETE_BEHAVIOR)
+class GooglePlacesStoreSearchAlgorithm implements StoreSearchAlgorithm<Place>
+{
+    
+    private final static Logger LOG = LoggerFactory.getLogger(GooglePlacesStoreSearchAlgorithm.class);
+    static final int DEFAULT_RADIUS = 1_000;
```

Comment:
Sometimes the data is way off when it comes to location. Might be a good idea to increase this up to 3,000-5,000.